homeassistant/components/dsmr/sensor.py

```diff
@@ -401,9 +508,18 @@ async def async_setup_entry(
         seconds=entry.options.get(CONF_TIME_BETWEEN_UPDATE, DEFAULT_TIME_BETWEEN_UPDATE)
     )
 
+    created = set()
+
     @Throttle(min_time_between_updates)
     def update_entities_telegram(telegram: dict[str, DSMRObject] | None) -> None:
-        """Update entities with latest telegram and trigger state update."""
+        """Add and update entities depending on the DSMR data we received."""
+        mbus_entities = create_mbus_entities(telegram, entry)
+        for mbus_entity in mbus_entities:
+            if mbus_entity.unique_id not in created:
+                created.add(mbus_entity.unique_id)
+                async_add_entities([mbus_entity])
```

Comment:
```suggestion
                new_entities.append(mbus_entity)
```

Please call  `async_add_entities` only ones.